geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -604,8 +604,9 @@ protected LocalRegion(String regionName, RegionAttributes attrs, LocalRegion par
     subregions = new ConcurrentHashMap();
 
     if (internalRegionArgs.getCachePerfStatsHolder() != null) {
-      hasOwnStats = false;
-      cachePerfStats = internalRegionArgs.getCachePerfStatsHolder().getCachePerfStats();
+      HasCachePerfStats cachePerfStatsHolder = internalRegionArgs.getCachePerfStatsHolder();
+      hasOwnStats = cachePerfStatsHolder.hasOwnStats();
+      cachePerfStats = cachePerfStatsHolder.getCachePerfStats();
```

Comment:
Please add tests showing that the LocalRegion closes cachePerfStats when the holder has own stats, and does not when the holder doesn't have own stats.

Reply:
I just pushed LocalRegionTest with two new tests. Please take a look!